version: '3.1'

services:
  app:
    image: mokhlesur96/dj:9ca75ce64a62b7f11929ab24747a47a95d154433
    container_name: py-depl
    #    build:
    #      context: .
    ports:
      - 8000:8000
    restart: always
    networks:
      - py-project


  database:
    image: postgres
    container_name: go-api-db-svc
    ports:
      - 5432:5432
    environment:
      - POSTGRES_DB=core
      - POSTGRES_USER=admin
      - POSTGRES_PASSWORD=secret
    volumes:
      - ${PWD}/database/data/:/var/lib/postgresql/data/
    networks:
      - py-project

  nginx:
    image: nginx
    container_name: nginx-proxy
    ports:
      - 80:80
      - 443:443
    volumes:
      - ./nginx.conf:/etc/nginx/conf.d/default.conf
      - ./ssl:/ssl
    depends_on:
      - app
    networks:
      - py-project

  prometheus:
    image: bitnami/prometheus:latest
    container_name: prometheus
    ports:
      - "9090:9090"
    restart: on-failure
    volumes:
      - ./prometheus.yml:/opt/bitnami/prometheus/conf/prometheus.yml
    command:
      - '--config.file=/opt/bitnami/prometheus/conf/prometheus.yml'
    depends_on:
      - app
    networks:
      - py-project


  node-exporter:
    image: bitnami/node-exporter:latest
    ports:
      - 9100:9100
    command:
      - '--path.rootfs=/hostfs'
      - '--collector.textfile.directory=/opt/bitnami/node-exporter'
    restart: always
    networks:
      - py-project

networks:
  py-project:
    driver: bridge


